src/com/google/javascript/jscomp/parsing/parser/util/UnicodeMatch.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp.parsing.parser.util;
+
+import java.util.regex.Pattern;
+
+/**
+ * Exposes non-ASCII unicode detection for Javascript identifiers in compliance
+ * with ECMAScript spec. Retains old optimizations for commonly-used characters,
+ * namely ASCII and Latin/Greek charsets. Uses regex for remaining characters
+ * (which would throw errors otherwise).
+ * 
+ * Unicode category patterns with /u flag compiled to vanilla regex with
+ * regexpu: https://github.com/mathiasbynens/regexpu
+ * 
+ * IdentifierStart and IdentifierPart regex patterns based on ECMA's
+ * documentation:
+ * https://github.com/tc39/proposal-regexp-unicode-property-escapes#other-examples
+ * 
+ * const regexIdentifierStart = /[$_\p{ID_Start}]/u;
+ * const regexIdentifierPart = /[$_\u200C\u200D\p{ID_Continue}]/u;
+ */
+public class UnicodeMatch {
```

Comment:
Some visibility comments:

This class should be final. I think it shouldn't be public, unless we later find a compelling use case outside this package.

The Google Java best practices also recommends adding an empty private constructor for utility classes to make it clear they should not be instantiated, i.e.

``` java
  private UnicodeMatch() {}
```

Reply:
Thank you very much Laura, I will make the relevant changes.

Reply:
If the class is made `private`, I can't import it inside `Scanner.java` - extremely sorry if this is basic syntax, I tried to research it, but is there any way to import *just* `public isJavascriptIdentifierStart` and `public isJavascriptIdentifierPart` from `private UnicodeMatch` without importing the whole class? I tried:
```java
import com.google.javascript.jscomp.parsing.parser.util.UnicodeMatch.isJavascriptIdentifierStart;
import com.google.javascript.jscomp.parsing.parser.util.UnicodeMatch.isJavascriptIdentifierPart;
```

But it will not build without importing the whole class (while `public`) via
```java
import com.google.javascript.jscomp.parsing.parser.util.UnicodeMatch
```

And referencing with `UnicodeMatch.isJavascriptIdentifierStart` and so on.